src/cpp/shared/spell/spells.cpp

```diff
@@ -159,7 +159,7 @@ int __fastcall GetManaCost(int spell, hero* hro) {
   int cost = GetManaCost_orig(spell, hro);
   auto res = ScriptCallbackResult<int>("OnCalcManaCost", deepbind<hero*>(hro), spell, cost);
   if(res.has_value())
-    cost = res.value();
+    cost = max(0, res.value());
```

Comment:
Y'know, a buttonn that eats up your spell-cast but givess you mana back sounnds like an interesting idea....

Reply:
I think something will break in the same way as mobility did

Reply:
Looks like it works, though obviously it shows -value instead of +value if you look in the magic book
Should I remove this check?